File: PHP/classes/Basic_Problems_Solutions.php
<?php
class Basic_Problems_Solutions {
	public $str;
	public $num;

	public function vowels_Consonents_Digits() {
		$n = strlen($this->str);
		$output = array();
		$output['input'] = $this->str;
		$res = array();
		for($i = 0; $i < $n; $i++) {
			$c = $this->str[$i];
			$d = ord($c);
			if($c == 'a' || $c == 'e' || $c == 'i' || $c == 'o' || $c == 'u' || $c == 'A' || $c == 'E' || $c == 'I' || $c == 'O' || $c == 'U')
				$res[$c] = 'vowel';
			else if($d >= 48 && $d <= 57)
				$res[$c] = 'digit';
			else
				$res[$c] = 'consonent';
		}
		$output['output'] = $res;
		return $output;
	}

	public function reverse_number() {
		$n = (int) $this->num;
		$r = 0;
		$d = 0;
		$output = array();
		$output['input'] = $n;
		
		while($n != 0) {
			$d = $n % 10;
			$r = $r * 10 + $d;
			$n = (int) ($n / 10);
		}
		$output['output'] = $r;
		return $output;
	}

	public function decimalToBinary() {
		$n = (int) $this->num;
		$i = 0;
		$binStr = '';
		$output = array();
		$output['input'] = $this->num;
		while($n > 0) {
			$bin = $n % 2;
			$n = (int) ($n / 2) ;
			$binStr .= $bin;
		}
		$output['output'] = $binStr;
		return $output;
	}

	public function find_next_20_leap_year() {
		$y = (int) $this->num;
		$i = 1;
		$output = array();
		$output['input'] = $this->num;
		$res = array();

		while( $i <= 20 ) {
			if( ($y % 4 == 0 && $y % 100 != 0) || $y % 400 == 0 ) {
				$res[$i] = "next leap year is ".$y;
				$i++;
			}
			$y++;
		}
		$output['output'] = $res;
		return $output;
	}
}
?>

File: Python3/Basic_Problems_Solutions.py
from colorama import Fore
class Basic_Problems_Solutions :
	def vowels_consonents_digits(self,str) :
		for i in str :
			if i == 'a' or i == 'e' or i == 'i' or i == 'o' or i == 'u' or i == 'A' or i == 'E' or i == 'I' or i == 'O' or i == 'U' :
				print(i,": vowel")
			elif ord(i) >= 48 and ord(i) <= 57 :
				print(i,": digit")
			else :
				print(i,": consonents")
	def reverse_number(self,num) :
		n,r,d = num,0,0
		while n != 0 :
			d = n % 10
			r = r * 10 + d
			n = int(n / 10)
		print("Reversed Number :",r)
	def decimalToBinary(self,num) :
		n = num
		bin = []
		while n > 0 :
			bin.append( n % 2 )
			n = int(n / 2)
		print("Binary Value of",num,": ",end="")
		bin = bin[::-1]
		for i in bin :
			print(i,end="")
		print()

a = Basic_Problems_Solutions()
print(Fore.RED + "\t Basic Problems and Solutions")
print(Fore.WHITE + " 1.Vowels, Consonets and Digits")
print(Fore.WHITE + " 2.Reverse a Number")
print(Fore.WHITE + " 3.Decimal to Binary Convertor");
print(Fore.GREEN + "\n Enter Your Choice : ",end="")
choice = int(input())
if choice == 1 :
	print(Fore.YELLOW + "\n\t Vowels, Consonents and Digits")
	vcd = input(Fore.GREEN + "Enter a String : ")
	a.vowels_consonents_digits(vcd)
elif choice == 2 :
	print(Fore.YELLOW + "\n\t Reverse a Number ")
	num = int(input(Fore.GREEN + "Enter Number : "))
	a.reverse_number(num)
elif choice == 3 :
	print(Fore.YELLOW + "\n\t Decimal to Binary")
	num = int(input(Fore.GREEN + "Enter Number : "))
	a.decimalToBinary(num)
else :
	print("Error : Invalid Choice");
